vowpalwabbit/cb_explore_adf_cover.cc

```diff
@@ -239,14 +278,27 @@ VW::LEARNER::base_learner* setup(config::options_i& options, vw& all)
   all.example_parser->lbl_parser = CB::cb_label;
   all.label_type = label_type_t::cb;
 
+  bool epsilon_decay;
+  if (options.was_supplied("epsilon"))
+  {
+    // fixed epsilon during learning
+    epsilon_decay = false;
+  }
+  else
+  {
+    epsilon = 1.f;
+    epsilon_decay = true;
+  }
+
   using explore_type = cb_explore_adf_base<cb_explore_adf_cover>;
-  auto data = scoped_calloc_or_throw<explore_type>(
-      cover_size, psi, nounif, first_only, as_multiline(all.cost_sensitive), all.scorer, cb_type_enum);
+  auto data = scoped_calloc_or_throw<explore_type>(cover_size, psi, nounif, epsilon, epsilon_decay, first_only,
+      as_multiline(all.cost_sensitive), all.scorer, cb_type_enum, all.model_file_ver);
 
   VW::LEARNER::learner<explore_type, multi_ex>& l = init_learner(
       data, base, explore_type::learn, explore_type::predict, problem_multiplier, prediction_type_t::action_probs);
 
   l.set_finish_example(explore_type::finish_multiline_example);
+  l.set_save_load(explore_type::save_load);
```

Comment:
Oh I see, this sets up the hook, so if you call this, you better have defined a class method.

Reply:
Exactly :D You would get a compiler error otherwise so it should be safe as it is